### PR TITLE
Feature/extract into more pkgs1

### PR DESCRIPTION
**Pull request by mvo5**
Build on top of the native-security branch, once thats in this branch diff is a lot smaller.

This branch moves the timeout into its own (tiny) package. The idea is to get it out of the way to move the security.go into its own package eventually. Doing that is a bit tricky because of the inter-dependencies between packageYaml and SecurityDefinition (used in ServicesYaml and Binaries) in the code right now.

### CODE REVIEW

**Comment by niemeyer** [comment]
Thanks, Michael. Please ping me once the alreayd reviewed top branch is in so this can be more easily reviewed.

**Comment by chipaca** [comment]
:+1:

File: timeout/timeout_test.go [hunk 1]
```diff
@@ -17,16 +17,25 @@
...
 	"time"
 
 	. "gopkg.in/check.v1"
 )
 
-func (s *SnapTestSuite) TestTimeoutMarshal(c *C) {
+// Hook up check.v1 into the "go test" runner
+func Test(t *testing.T) { TestingT(t) }
+
+type TimeoutTestSuite struct {
```
Comment (niemeyer) [inline timeout/timeout_test.go]: For future endeavors, empty structs are generally formatted as `struct{}`.

**Comment by niemeyer** [comment]
Small and sweet, thanks.

**Pull request merged**